huaweicloud/services/evs/resource_huaweicloud_evs_snapshot_group.go

```diff
@@ -0,0 +1,517 @@
+package evs
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var snapshotGroupNonUpdatableParams = []string{
+	"server_id",
+	"volume_ids",
+	"instant_access",
+	"enterprise_project_id",
+	"incremental",
+}
+
+// @API EVS POST /v5/{project_id}/snapshot-groups
+// @API EVS PUT /v5/{project_id}/snapshot-groups/{snapshot_group_id}
+// @API EVS GET /v5/{project_id}/snapshot-groups/{snapshot_group_id}
+// @API EVS DELETE /v5/{project_id}/snapshot-groups/{snapshot_group_id}
+// @API EVS POST /v5/{project_id}/snapshot-groups/{snapshot_group_id}/tags/create
+// @API EVS POST /v5/{project_id}/snapshot-groups/{snapshot_group_id}/tags/delete
+func ResourceEvsSnapshotGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceEvsSnapshotGroupCreate,
+		ReadContext:   resourceEvsSnapshotGroupRead,
+		UpdateContext: resourceEvsSnapshotGroupUpdate,
+		DeleteContext: resourceEvsSnapshotGroupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		CustomizeDiff: config.FlexibleForceNew(snapshotGroupNonUpdatableParams),
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"server_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"volume_ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"instant_access": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"description": {
```

Comment:
If you add a computed attribute, the description field cannot be empty.
It is recommended to remove the computed attribute and add a test to verify that the description field is edited from a value to an empty value in the test case.